### PR TITLE
feat: Add statefulset user guide

### PR DESCRIPTION
**Pull request by harrisonliu5**
Signed-off-by: Harrison harrisonliu_5@163.com

### CODE REVIEW

**Comment by leoendless** [comment]
Great work! I think the only short is the documentation of resource operations needs to be enriched.

File: content/en/docs/project-user-guide/application-workloads/statefulsets.md [hunk 1]
```diff
@@ -7,4 +7,299 @@ description: 'Kubernetes StatefulSets'
...
+
+  The Partitions is mean which pods you want to update through the rolling update. It specifying by `.spec.updateStrategy.rollingUpdate.partition`. The default is 0. For more information about Partitions, please visit [Partitions](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#partitions).
+
+- **Pod Security Context**
+
+  A security context defines privilege and access control settings for a Pod or Container. For more information about Pod Security Policies, please visit [Pod Security Policies](https://kubernetes.io/docs/concepts/policy/pod-security-policy/).
+
+- **Deployment Mode**
+
+  Select different deployment modes to change the affinity and anti-affinity of the inter-pod. In k8s, the affinity is used `podAffinity` and the anti-affinity is used `podAntiAffinity`. In the Kubesphere both affinity and anti-affinity are set `preferredDuringSchedulingIgnoredDuringExecution`.
```
Comment (Sherlock113) [inline content/en/docs/project-user-guide/application-workloads/statefulsets.md:199]: ->
You can select different deployment modes to switch between inter-pod affinity and inter-pod anti-affinity. In Kubernetes, inter-pod affinity is specified as field `podAffinity` of field `affinity` while inter-pod anti-affinity is specified as field `podAntiAffinity` of field `affinity`. In KubeSphere, both `podAffinity` and `podAntiAffinity` are set to `preferredDuringSchedulingIgnoredDuringExecution`. You can enable **Edit Mode** in the top right corner to see field details.

File: content/en/docs/project-user-guide/application-workloads/statefulsets.md [hunk 2]
```diff
@@ -7,4 +7,299 @@ description: 'Kubernetes StatefulSets'
...
+
+For more information about volume, please visit [Volumes](../../storage/volumes)
+
+### Step 5: Input Advanced Data
+
+![statefulsets](/images/docs/project-user-guide/workloads/statefulsets_form_4.png)
+
+- **Set Node Scheduling Policy**
+
+  You can allow Pod replicas to run on specified nodes. It is used `nodeSelector`.
```
Comment (Sherlock113) [inline content/en/docs/project-user-guide/application-workloads/statefulsets.md:250]: It is used `nodeSelector`.
->
It is specified in the field `nodeSelector`.

**Review by Sherlock113 — COMMENTED** [review]
Please have a look at my comments and make changes. Thanks.

**Comment by Sherlock113** [comment]
I am actually thinking about extracting part of the content in this guide to a separate guide. This guide covers detailed explanations of some shared attributes which can also be used in the guide of other workloads.

I think we can just put these explanations in this guide for now, and I will refactor some of the content when all guides of this kind are finished.

**Comment by Sherlock113** [comment]
/lgtm
@FeynmanZhou Please merge this pr and I will do the rest. Thanks.

**Comment by FeynmanZhou** [comment]
> I am actually thinking about extracting part of the content in this guide to a separate guide. This guide covers detailed explanations of some shared attributes which can also be used in the guide of other workloads.
> 
> I think we can just put these explanations in this guide for now, and I will refactor some of the content when all guides of this kind are finished.

@Sherlock113 Please also open the related GitHub issue to track your plans.

**Pull request merged**